autogen/agentchat/contrib/gpt_assistant_agent.py

```diff
@@ -52,12 +52,28 @@ def __init__(
                 - verbose (bool): If set to True, enables more detailed output from the assistant thread.
                 - Other kwargs: Except verbose, others are passed directly to ConversableAgent.
         """
+
+        self._verbose = kwargs.pop("verbose", False)
+        super().__init__(
+            name=name, system_message=instructions, human_input_mode="NEVER", llm_config=llm_config, **kwargs
+        )
+
+        if llm_config is False:
+            raise ValueError("llm_config=False is not supported for GPTAssistantAgent.")
+
         # Use AutoGen OpenAIWrapper to create a client
         openai_client_cfg = None
-        model_name = "gpt-4-1106-preview"
-        if llm_config and llm_config.get("config_list") is not None and len(llm_config["config_list"]) > 0:
+        if llm_config.get("config_list") is not None and len(llm_config["config_list"]) > 0:
             openai_client_cfg = llm_config["config_list"][0].copy()
-            model_name = openai_client_cfg.pop("model", "gpt-4-1106-preview")
+        else:
+            openai_client_cfg = llm_config.copy()
+            openai_client_cfg.pop("config_list", None)
+
+        if openai_client_cfg is None:
+            raise ValueError("OpenAI client config is not found in llm_config.")
+
+        logger.warning("OpenAI client config of GPTAssistantAgent(%s): %s", name, openai_client_cfg)
+        model_name = openai_client_cfg.pop("model", "gpt-4-1106-preview")
 
         oai_wrapper = OpenAIWrapper(**openai_client_cfg)
```

Comment:
- Line 67 and line 78 together are problematic. `llm_config` may contain args that are needed by `OpenAIWrapper`. I suggest passing a copy of `llm_config` with "model" popped in line 78.
- Line 70 is unnecessary.
- Line 75 has the risk of exposing sensitive info such as api keys and endpoints. I suggest removing it or only log the model and api_type.
- Line 76: what about using "0125" instead of "1106" as the default model version?

Reply:
Good reminder, that was indeed the plan to make such changes, but it slipped my mind as I was making them.

Reply:
PTAL again